nx/include/switch/applets/error.h

```diff
@@ -0,0 +1,76 @@
+/**
+ * @file error.h
+ * @brief Wrapper for using the error LibraryApplet.
+ * @author StuntHacks
+ * @copyright libnx Authors
+ */
+#pragma once
+#include "../types.h"
+#include "../services/applet.h"
+
+/// Error configuration struct.
+typedef struct {
+    bool custom_text;                 ///< Whether to show a custom error message. If this is false, a default message will be shown.
+    u8 pad[7];
+    u32 major_code;                   ///< First part of the error-code.
+    u32 minor_code;                   ///< Second part of the error-code.
+    u8 pad2[8];
+    char short_description[0x800];    ///< Short description.
+    char detailed_description[0x800]; ///< Detailed description (displayed when the user clicks,  on "Details").
+} ErrorConfig;
+
+/**
+ * @brief Creates an ErrorConfg struct.
+ * @param c ErrorConfg struct.
+ * @note Sets the following fields: majorCode = 2000, minorCode = 0, customText = false, shortDescription = "", detailedDescription = "".
```

Comment:
Add big fat `@warning` saying that this applet creates an error report that is logged in the system and that the user should proceed at their own risk.